dev-libs/glib: add 2.82.2 and related

PeakKS opened the pull request:
dev-libs/gobject-introspection-common: add 1.82.0
dev-libs/gobject-introspection: add 1.82.0
dev-libs/glib: add 2.82.2
dev-util/glib-utils: add 2.82.2
dev-util/gdbus-codegen: add 2.82.2
dev-python/pygobject: add 3.50.0
sys-libs/libmodulemd: replace gtk-doc use for glib
media-video/aravis: replace gtk-doc use for glib


Contains potential fix for https://bugs.gentoo.org/937616:
- Uses an internal build of gobject-introspection to build glib with introspection enabled

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

PeakKS commented:
Should be good now, tested with multiib/nomultilib, as well as upgrading from 2.78. The internal gobject-introspection is only used for building glib, not actually installed, so gobject-introspection can still be freely bumped when needed.

PeakKS commented:
Ok I have no idea what the qa bot is complaining about at this point

Dessa commented on dev-util/gdbus-codegen/gdbus-codegen-2.82.0.ebuild:
```diff
@@ -0,0 +1,80 @@
+RDEPEND="
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		dev-python/packaging[${PYTHON_USEDEP}]
```
appears to be no longer required as per https://gitlab.gnome.org/GNOME/glib/-/commit/38faeca62ebd5be89b1ec3e6c530e499ce2d7240

PeakKS replied:
Thanks for finding, dropped

eli-schwartz commented:
This PR looks quite exciting. Small thoughts:

- When building the private introspection data, maybe it would be possible to check if we `has_version` a good enough copy already installed, and skip building a private one?
- the commit message for `dev-libs/glib: add 2.82.1` should be more detailed, and explain what the commit does other than just bump the version. This is very useful information when running `git log`, so please provide it :)

eli-schwartz commented on dev-libs/glib/glib-2.82.1.ebuild:
```diff
@@ -0,0 +1,407 @@
+		ORIG_BUILD_DIR=${BUILD_DIR}
+		BUILD_DIR=${INTROSPECTION_BUILD_DIR}
+
+		pushd ${INTROSPECTION_SOURCE_DIR}
```
pushd / popd aren't PMS functions so you do have to guard them with `|| die` according to policy.

eli-schwartz commented on dev-libs/glib/glib-2.82.1.ebuild:
```diff
@@ -0,0 +1,407 @@
+
+		pushd ${INTROSPECTION_SOURCE_DIR}
+
+		meson_src_configure
+		meson_src_compile
+		# Install to the portage temp directory so that pkgconfig relative paths resolve correctly
+		# The install directory will get wiped by portage before the actual install stage
+		meson_src_install --skip-subprojects glib
```
Does this mean we are actually accessing ${D} in src_configure... maybe it would be better to customize the eclass usage in order to install in ${T} instead.

eli-schwartz commented on dev-libs/glib/glib-2.82.1.ebuild:
```diff
@@ -0,0 +1,407 @@
+	# Build internal copy of gobject-introspection to avoid circular dependency (Built for native abi only)
+	if multilib_native_use introspection; then
+		local emesonargs=(
+			$(meson_feature test cairo)
```
I don't think we need this at all since we're not running tests for the internal copy of gobject-introspection, are we?

PeakKS replied:
Yeah shouldn't be, will drop

pacho2 commented:
It would also be interesting to have a similar one for 2.80.x version (that will probably go with Gnome 46, while glib 2.82 will probably stay with Gnome 47)
Thanks

PeakKS commented:
> This PR looks quite exciting. Small thoughts:
> 
>     * When building the private introspection data, maybe it would be possible to check if we `has_version` a good enough copy already installed, and skip building a private one?

This seems to work well, I added a has_version check after the meson_native_use check.

>     * the commit message for `dev-libs/glib: add 2.82.1` should be more detailed, and explain what the commit does other than just bump the version. This is very useful information when running `git log`, so please provide it :)

I added a bit more information to it.

pacho2 commented:
> It would also be interesting to have a similar one for 2.80.x version (that will probably go with Gnome 46, while glib 2.82 will probably stay with Gnome 47) Thanks

Thanks a lot for your work.

I wonder if maybe updating to 2.82.2 would be possible (to simply merge https://github.com/gentoo/gentoo/pull/39090 here and do everything at the same time)

Best regards

PeakKS commented:
> > It would also be interesting to have a similar one for 2.80.x version (that will probably go with Gnome 46, while glib 2.82 will probably stay with Gnome 47) Thanks
> 
> Thanks a lot for your work.
> 
> I wonder if maybe updating to 2.82.2 would be possible (to simply merge #39090 here and do everything at the same time)
> 
> Best regards

Was able to update to 2.82.2 without issues, I'll look into making another PR adapting 2.80 as well.

Norbiros commented:
I think this shouldn't be closed
